core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,96 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.slf4j.{Marker, MarkerFactory}
 
-trait Logging {
-  val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+import scala.util.{Failure, Success, Try}
 
-  protected var logIdent: String = null
+object Log4jControllerRegistration {
+  private val logger = Logger(this.getClass.getName)
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  val FatalMarker: Marker = MarkerFactory.getMarker("FATAL")
 
-  protected def msgWithLogIdent(msg: String) =
-    if (logIdent == null) msg else logIdent + msg
-
-  def trace(msg: => String): Unit = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg))
-  }
-  def trace(e: => Throwable): Any = {
-    if (logger.isTraceEnabled())
-      logger.trace(logIdent,e)
+  val instance = Try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
```

Comment:
Why is the `asInstanceOf` necessary?

Reply:
Well, I think the compiler is not able to figure out what is the return type of the Class.forName call and guesses it's Class<Any>, therefore I get type mismatch at CoreUtils.registerMBean:
```
type mismatch;
 found   : Any
 required: Object
    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
```
Also I can't do `.asInstanceOf[Class[Log4jController]]` as it is private and so is the `Log4jControllerMBean`. I didn't want to change the visibility of these classes from private as we're about to throw them out.